### PR TITLE
Give each report builder report its own data source

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Previously, we tried to share data sources between report builder reports in an attempt to save storage space. However, this resulted in a very annoying layer of complexity which seems unnecessary, makes future features harder to implement. This PR ends that sharing.

cc @emord 
FYI @kaapstorm 
Can be reviewed commit-by-commit

### CODE REVIEW

File: corehq/apps/userreports/tests/test_report_builder.py [hunk 1]
```diff
@@ -29,10 +30,97 @@ def tearDownClass(cls):
...
+            }
+        )
+        self.assertTrue(builder_form.is_valid())
+        report_two = builder_form.create_report()
+
+        self.assertNotEqual(report_one.config_id, report_two.config_id)
+
+    def test_updating_report_data_source(self):
+        """
+        Test that changing the app or number column for a report results in an update to the data source next time the
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_report_builder.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/userreports/tests/test_report_builder.py [hunk 2]
```diff
@@ -29,10 +30,97 @@ def tearDownClass(cls):
...
+        builder_form = ConfigureTableReportForm(
+            "Test Report",
+            self.app._id,
+            "case",
+            "some_case_type",
+            existing_report=report,
+            data={
+                'group_by': 'closed',
+                'filters': '[]',
+                # Note that a "Sum" calculation on the closed case property isn't very sensical, but doing it so that
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_report_builder.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/userreports/tests/test_report_builder.py [hunk 3]
```diff
@@ -29,10 +30,97 @@ def tearDownClass(cls):
...
+                # I can have a numeric calculation without having to create real case properties for this case type.
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_report_builder.py]: E501 line too long (116 > 115 characters)

File: corehq/apps/userreports/tests/test_report_builder.py [hunk 4]
```diff
@@ -29,10 +30,97 @@ def tearDownClass(cls):
...
+                'columns': '[{"property": "closed", "display_text": "closed", "calculation": "Sum"}]',
+            }
+        )
+        self.assertTrue(builder_form.is_valid())
+        builder_form.update_report()
+
+        # reload report data source, because report.config is memoized
+        data_source = DataSourceConfiguration.get(report.config._id)
+        # The closed property indicator should now be decimal type because the user indicated that it was numeric by
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_report_builder.py]: E501 line too long (116 > 115 characters)

**Comment by benrudolph** [comment]
fan of this!!

**Comment by kaapstorm** [comment]
:+1:

**Pull request merged**